### ISSUE TITLE
Kubernetes memory consumption explosion

### ISSUE THREAD

**Issue by RenaudWasTaken** [kubernetes/kubernetes#43593]
**Is this a BUG REPORT or FEATURE REQUEST?**: Bug report

**What happened**:
TLDR: kubelet explodes in RAM on a 32GB machine
Current master: 03aba86974957feae11a73a0db43a39b5743c550
Date:   Thu Mar 23 12:45:38 2017 -0700

1) git clone https://github.com/kubernetes/kubernetes.git (
2) cd ./kubernetes && ./hack/local-up-cluster.sh
3) open sudo htop and wait 4-5 minutes on a 32GB machine
4) Observe the insane amount of memory consumed

**What you expected to happen**:
Not 32Gb of memory consumed

**Anything else we need to know**:
This seem to be a bug in kubelet as it seem to be trying to mount volumes ?
```bash
$ tail /tmp/kubelet.log
E0323 14:04:47.944598   10536 kubelet_volumes.go:114] Orphaned pod "1347d388-0ded-11e7-8127-38d547255e91" found, but volume paths are still present on disk.
I0323 14:04:48.571515   10536 kubelet_volumes.go:104] Orphaned pod "18005e0b-0f68-11e7-9e19-38d547255e91" found, but volumes are not cleaned up
I0323 14:04:48.708207   10536 server.go:235] Checking API server for new Kubelet configuration.
I0323 14:04:48.839060   10536 server.go:244] Did not find a configuration for this Kubelet via API server: cloud provider was nil, and attempt to use hostname to find config resulted in: configmaps "kubelet-127.0.0.1" not found
I0323 14:04:49.217504   10536 kubelet_volumes.go:104] Orphaned pod "192e8eba-0c55-11e7-b36b-38d547255e91" found, but volumes are not cleaned up
E0323 14:04:49.382762   10536 kubelet_volumes.go:114] Orphaned pod "1ce85d54-0f40-11e7-b0f1-38d547255e91" found, but volume paths are still present on disk.
E0323 14:04:49.666222   10536 kubelet_volumes.go:114] Orphaned pod "1e910bc5-0deb-11e7-a21f-38d547255e91" found, but volume paths are still present on disk.
I0323 14:04:49.987318   10536 kubelet_volumes.go:104] Orphaned pod "1f6faf7c-0c55-11e7-b36b-38d547255e91" found, but volumes are not cleaned up
E0323 14:04:50.125429   10536 kubelet_volumes.go:114] Orphaned pod "358df142-0f6a-11e7-9c91-38d547255e91" found, but volume paths are still present on disk.
E0323 14:04:50.263115   10536 kubelet_volumes.go:114] Orphaned pod "44698294-0e78-11e7-99ac-38d547255e91" found, but volume paths are still present on disk.
E0323 14:04:50.400227   10536 kubelet_volumes.go:114] Orphaned pod "484233df-0df2-11e7-ba67-38d547255e91" found, but volume paths are still present on disk.
E0323 14:04:50.537138   10536 kubelet_volumes.go:114] Orphaned pod "4bbfd232-0f3f-11e7-88f0-38d547255e91" found, but volume paths are still present on disk.
I0323 14:04:50.675202   10536 kubelet_volumes.go:104] Orphaned pod "4d376830-0f68-11e7-a715-38d547255e91" found, but volumes are not cleaned up
E0323 14:04:50.812823   10536 kubelet_volumes.go:114] Orphaned pod "52f69643-0df2-11e7-ba67-38d547255e91" found, but volume paths are still present on disk.
E0323 14:04:50.952176   10536 kubelet_volumes.go:114] Orphaned pod "58adb888-0f3b-11e7-a7bd-38d547255e91" found, but volume paths are still present on disk.
E0323 14:04:51.143531   10536 kubelet_volumes.go:114] Orphaned pod "6fed98a0-0e93-11e7-8e3a-38d547255e91" found, but volume paths are still present on disk.
E0323 14:04:51.351032   10536 kubelet_volumes.go:114] Orphaned pod "7ea81fc0-0f6a-11e7-a2ba-38d547255e91" found, but volume paths are still present on disk.
E0323 14:04:51.512311   10536 kubelet_volumes.go:114] Orphaned pod "811f82eb-0f67-11e7-bf73-38d547255e91" found, but volume paths are still present on disk.

```

**Environment and version**:

```bash
$ cluster/kubectl.sh version
Client Version: version.Info{Major:"1", Minor:"7+", GitVersion:"v1.7.0-alpha.0.1479+03aba86974957f-dirty", GitCommit:"03aba86974957feae11a73a0db43a39b5743c550", GitTreeState:"dirty", BuildDate:"2017-03-23T21:54:07Z", GoVersion:"go1.8", Compiler:"gc", Platform:"linux/amd64"}

$ cat /etc/os-release                                                                                                                                        NAME="Ubuntu"
VERSION="16.04.2 LTS (Xenial Xerus)"
PRETTY_NAME="Ubuntu 16.04.2 LTS"
VERSION_ID="16.04"
UBUNTU_CODENAME=xenial

$ uname -a                                                                                                                                                   
Linux 4.4.0-62-generic #83-Ubuntu SMP x86_64 x86_64 x86_64 GNU/Linux
```
Ran from `./hack/local-up-cluster.sh`

**Comment by RenaudWasTaken** [comment]
Cleaning up kubernetes files in the environment seems to have fixed it

**Comment by jayunit100** [comment]
I wonder wether it would have self-healed eventually ?

**Comment by rangapv** [comment]
Hi,
I am getting the following error when trying to start a pod via the single node K8 with CRI-O in ubuntu16.1, the hardware is 4 vCPU with 10 GB RAM on GCP. Error message is failed to sync pod, error creating the pod, CrashLoopBackOff.... The following is the POD description error message.....

 52s   5s      6       kubelet, 127.0.0.1      spec.containers{nginx6} Warning BackOff         Back-off restarting 
failed container
  5s    5s      1       kubelet, 127.0.0.1                              Warning FailedSync      Error syncing pod, s
kipping: failed to "StartContainer" for "nginx6" with CrashLoopBackOff: "Back-off 40s restarting failed container=ng
inx6 pod=nginx6-1780739566-twrlx_default(f6cd200a-1d49-11e7-ac07-42010a800002)"

The following  is the kubelet.log error message...

4844 kubelet.go:1667] Failed to check if disk space is available for the runtime: failed to get fs info for "runtime": ImagesFsInfo: unknown runtime: remote


Thanks
Ranga
